alibi_detect/od/sklearn/svm.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from typing import Dict, Union, Callable, Optional
+from typing_extensions import Literal
+from sklearn.linear_model import SGDOneClassSVM
+
+from alibi_detect.od.sklearn.base import SklearnOutlierDetector
+from sklearn.kernel_approximation import Nystroem
+
+
+class SVMSklearn(SklearnOutlierDetector):
+    ensemble = False
+
+    def __init__(
+        self,
+        n_components: Optional[int] = None,
+        kernel: Union[Literal['linear', 'poly', 'rbf', 'sigmoid'], Callable] = 'rbf',
+        sigma: Optional[float] = None,
+        kernel_params: Optional[Dict] = None,
+    ):
+        """SKlearn backend for the One class support vector machine (SVM) outlier detector.
+
+        Parameters
+        ----------
+        n_components
+            Number of features to construct. How many data points will be used to construct the mapping.
+        kernel
+            Kernel map to be approximated by the Nystroem method. A callable should accept two arguments and the
+            keyword arguments passed to this object as ``kernel_params``, and should return a floating point number.
+            Otherwise the user can pass a string and the corresponding built-in kernel map will be used from
+            `sklearn.metrics.pairwise <https://scikit-learn.org/stable/modules/classes.html#pairwise-metrics/>`_.
+            See also
+            `sklearn.kernel_approximation.Nystroem <https://scikit-learn.org/stable/modules/generated/sklearn.kernel_approximation.Nystroem.html/>`_  # noqa
+        sigma
+            Sigma parameter for the RBF, polynomial, and sigmoid kernels. Interpretation of the default value is left
+            to the kernel; see the documentation for
+            `sklearn.metrics.pairwise <https://scikit-learn.org/stable/modules/classes.html#pairwise-metrics/>`_.
+            Ignored by other kernels.
+        kernel_params
+            Additional parameters (keyword arguments) for kernel function passed as callable object.
+
+        Raises
+        ------
+        ValueError
+            If `n_components` is less than 1.
```

Comment:
Not sure I see where this can be raised?

Reply:
Ahh, forgot to add this :facepalm: Added it now!

Reply:
👍🏻